hawkular-alerts-rest/src/main/java/org/hawkular/alerts/rest/EventsHandler.java

```diff
@@ -97,7 +96,11 @@ public Response createEvent(
                 if (null != alertsService.getEvent(tenantId, event.getId(), true)) {
                     return ResponseUtil.badRequest("Event with ID [" + event.getId() + "] exists.");
                 }
-                alertsService.addEvents(Collections.singletonList(event));
+                /*
+                    New events are sent directly to the engine for inference process.
+                    Input events and new ones generated by the alerts engine are persisted at the end of the process.
+                 */
+                alertsService.sendEvent(event);
```

Comment:
Yes.
We have logic to persist events (without sending to the engine) and send events (that checks active dataId and sends to the engine if necessary).
For me better approach could be to have only 1 method with the logic that sendEvent does, and make the addEvents private.